### DISCUSSION TITLE
DSL inside Window constructor

### DISCUSSION THREAD

**Discussion by Divelix** [libktx/ktx#342] [Q&A] [answered]
I just wanted to do the thing from #306 with `Window`, but it doesn't work this way as `KWindow` is a final class. How can I use ktx DSL inside class inherited from `Window`?

For example, this doesn't work:
```kotlin
class MyWindow(...) : Window(...), KWindow { ... }
```

**Comment by czyzby** [comment, marked as answer]
This is because `KWindow` is a concrete class of a widget, while `KTable` (from the linked example) is an interface that adds the DSL. Windows are actually just fancy tables in LibGDX - if you look into the `KWindow` implementation, you'll see that it also implements the `KTable` interface:

```kotlin

@Scene2dDsl
class KWindow(title: String, skin: Skin, style: String) : Window(title, skin, style), KTable
```

Instead of extending `KWindow`, all you have to do is implement `KTable` to get access to the DSL. When in doubt, check out which interfaces are implemented by the KTX widget. You can basically copy-and-paste the class and go from there.
